src/main/scala/com/fulcrumgenomics/fastq/FastqWriter.scala

```diff
@@ -37,16 +37,16 @@ object FastqWriter {
   def apply(path: Path): FastqWriter = apply(Io.toWriter(path))
 
   /** Constructs a FastqWriter from a Writer. */
-  def apply(writer: java.io.Writer): FastqWriter = {
-    if (writer.isInstanceOf[BufferedWriter]) new FastqWriter(writer.asInstanceOf[BufferedWriter])
-    else new FastqWriter(new BufferedWriter(writer))
+  def apply(writer: java.io.Writer): FastqWriter = writer match {
```

Comment:
Apply takes a writer and either makes sure it is a BufferedWriter or turns it into one. Is there any reason not to have this just require a BufferedWriter?